rllib/optimizers/async_replay_optimizer.py

```diff
@@ -206,16 +206,25 @@ def _step(self):
 
         with self.timers["sample_processing"]:
             completed = list(self.sample_tasks.completed())
-            counts = ray_get_and_free([c[1][1] for c in completed])
+
+            ray_error = None
             for i, (ev, (sample_batch, count)) in enumerate(completed):
-                sample_timesteps += counts[i]
+                # Task may have failed, try each without abandoning other,
+                # still good ones.
+                try:
+                    counts = ray_get_and_free(count)
+                except RayError as e:
```

Comment:
This will significantly degrade performance if the get is no longer batched. Is there another way of fixing this?

Reply:
I thought so, too. The problem is the ray.get([some IDs of good tasks, some IDs of crashed tasks]) throwing an error and then we can't capture the "returns" of the still-good tasks.

Reply:
This must be a general problem. I mean when we do `ray.get([ID1, ID2])` and ID2 has crashed, but we still care about ID1, what do we usually do?

Reply:
We could (1) retry with individual gets, or (2) re-dispatch tasks for the rest and drop the results.

Reply:
Changed: Trying batched first, then - only if that fails - single ray.get() to separate the errored tasks from the good ones w/o actually throwing an error before the good tasks have been kicked off again.